tdc/single_pred/develop.py

```diff
@@ -34,4 +34,96 @@ def __init__(self, name, path='./data', label_name=None, print_stats=False, conv
         self.entity1_name = 'Antibody'
         if print_stats:
             self.print_stats()
-        print('Done!', flush = True, file = sys.stderr)
\ No newline at end of file
+        print('Done!', flush = True, file = sys.stderr)
+        
+        
+    def graphein(self, graph = 'distance', node_feature = ['amino_acid_one_hot'], distance_threshold = 6, config = None, convertor = None):
+        from typing import Dict
+        import torch
+        from functools import partial
+
+        import graphein.protein as gp
+        from graphein.ml.conversion import GraphFormatConvertor
+        from graphein.ml import InMemoryProteinGraphDataset, ProteinGraphDataset
+        
+        from graphein.protein.utils import get_obsolete_mapping
+        import warnings
+        warnings.filterwarnings("ignore")
+        import logging
+        logging.getLogger("graphein").setLevel("ERROR")
+        try:
+            split = self.split
+        except:
+            raise ValueError('Please define data.split() first!')
+        
+        print_sys("Note that this task has obsolete PDBs, thus, we remove it in default. For fair comparison using other models, please retrieve the data.split object for the post removal splits!")
+        
+        from graphein.protein.utils import get_obsolete_mapping
+
+        obs = get_obsolete_mapping()        
+        train_obs = [t for t in split["train"]["Antibody_ID"] if t in obs.keys()]
+        valid_obs = [t for t in split["valid"]["Antibody_ID"] if t in obs.keys()]
+        test_obs = [t for t in split["test"]["Antibody_ID"] if t in obs.keys()]
+        
+        split["train"] = split["train"].loc[~split["train"]["Antibody_ID"].isin(train_obs)]
```

Comment:
Might want to add this for validation. There are no obs structures in the validation (at the moment!) but this may change.